packages/gatsby-plugin-offline/src/gatsby-node.js

```diff
@@ -0,0 +1,50 @@
+const Promise = require(`bluebird`)
+const precache = require(`sw-precache`)
+const path = require(`path`)
+
+exports.createPages = () => [
+  {
+    path: `/offline-plugin-app-shell-fallback/`,
+    component: path.resolve(`${__dirname}/app-shell.js`),
+  },
+]
+
+exports.postBuild = () => new Promise((resolve, reject) => {
+  const rootDir = `public`
+
+  const options = {
+    staticFileGlobs: [
+      `${rootDir}/**/*.{js,woff2}`,
+      `${rootDir}/index.html`,
+      `${rootDir}/manifest.json`,
+      `${rootDir}/offline-plugin-app-shell-fallback/index.html`,
+    ],
+    stripPrefix: rootDir,
+    navigateFallback: `/offline-plugin-app-shell-fallback/index.html`,
+    cacheId: `gatsby-plugin-offline`,
+    dontCacheBustUrlsMatching: /(.*.woff2|.*.js)/,
+    runtimeCaching: [
```

Comment:
`urlPattern` will trigger on partial URL matches, so there's no need to put in the leading `.*` wildcard.

If you wanted to combine all of the RegExps, you could do `/\.(?:png|jpg|jpeg)$/`

If you want to keep them separate, then I'd just tweak each of them a bit to escape that `.` character, and enforce that the match happens at the end of the URL string. So, e.g., `/\.png$/`.